### PR TITLE
Dividend feature under admin panel

### PR DESCRIPTION
**Pull request by Meivenkatkumar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "build:proto": "bash build_proto.sh",
     "build:webpack": "webpack -p",
-    "dev-server": "webpack-dev-server --colors --watch --hot --inline --port 8080 --host 0.0.0.0 --output-public-path=https://localhost:8080/build/ --https --cert=./public/tls_keys/dev/server.crt --key=./public/tls_keys/dev/server.key"
+    "dev-server": "webpack-dev-server --colors --watch --hot --inline --port 8080 --host 0.0.0.0 --output-public-path=http://localhost:8080/build/ "
```
[the remark this answers is not in the document]
Comment (Meivenkatkumar) [reply package.json]: There was an issue with https connections. So i tried with http.
The above changes may not be required in production env

File: public/src/components/App.tsx [hunk 2]
```diff
@@ -46,8 +46,8 @@ export class App extends React.Component<{}, AppState> {
 		// DalalActionService.serviceURL = "https://139.59.47.250";
 		// DalalStreamService.serviceURL = "https://139.59.47.250";
 
-		DalalActionService.serviceURL = "https://0.0.0.0:8000";
-		DalalStreamService.serviceURL = "https://0.0.0.0:8000";
+		DalalActionService.serviceURL = "http://0.0.0.0:8000";
```
[the remark this answers is not in the document]
Comment (Meivenkatkumar) [reply public/src/components/App.tsx]: In the next commit, i will change http->https

File: public/src/components/admin/Admin.tsx [hunk 3]
```diff
@@ -2,15 +2,71 @@ import * as React from "react";
...
+    dividendAmount: number
+}
+
+export class Admin extends React.Component<AdminProps,AdminState> {
     constructor(props: AdminProps) {
         super(props);
+        const currentStockId = Number(Object.keys(this.props.stockBriefInfoMap).sort()[0]);
+        this.state = {
+            currentStockId: currentStockId,
+            dividendAmount: 1
```
Comment (ishanisri) [inline public/src/components/admin/Admin.tsx]: Set dividend amount as 0

File: public/src/components/admin/Admin.tsx [hunk 4]
```diff
@@ -1,16 +1,108 @@
...
+        
+        if(isPositiveInteger(dividendAmount+1)){
+        const dividendReq = new SendDividendsRequest();
+        try{
+            dividendReq.setDividendAmount(dividendAmount);
+            dividendReq.setStockId(dividendStockId);
+            const resp = await DalalActionService.sendDividends(dividendReq, sessionMd);
+            // If any error occurs, it will be raised in DalalMessage_pb_Service
+            showNotif("Dividend has been successfully sent on "+stockName+"!");
+            console.log("Dividend request: StockId:"+this.state.currentStockId+", DividendAmount:"+this.state.dividendAmount);
```
Comment (ishanisri) [inline public/src/components/admin/Admin.tsx]: Remove console.log

File: public/src/components/admin/Admin.tsx [hunk 5]
```diff
@@ -22,11 +114,31 @@ export class Admin extends React.Component<AdminProps, {}> {
...
-            </div>
+            <React.Fragment>
+                <div className="adminPanel">
+                  <h1>ADMIN PANEL</h1>
+                  <span>dsaaaaaaaaaaaaaaaaaaaaaaaa</span><button onClick = {(e) => { this.purchaseFromExchange(e) }}>Send</button>       
+                </div>
+
+                {/* Dividend Feature */}
+                
+                <div className="dividendPanel" style={{ float: 'right' }}>
```
Comment (ishanisri) [inline public/src/components/admin/Admin.tsx]: All css has to be written in dalal.css

File: public/src/components/admin/Admin.tsx [hunk 6]
```diff
@@ -22,11 +114,31 @@ export class Admin extends React.Component<AdminProps, {}> {
...
+                  <Dividend 
+                   sessionMd={this.props.sessionMd} 
+                   stockBriefInfoMap={this.props.stockBriefInfoMap}
+                   currentStockId={this.state.currentStockId}
```
Comment (ishanisri) [inline public/src/components/admin/Admin.tsx]: No need to make `currentStockId` and `dividendAmount` as props, they can be state variables inside Dividend. You can simplify the functions also accordingly.

File: public/src/components/admin/Dividend.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        const stockBriefInfoMap = this.props.stockBriefInfoMap;
+        const options = [];
+        for (const stockId in stockBriefInfoMap) {
+            const stockInfo = stockBriefInfoMap[stockId];
+            options.push(
+                      <option value={stockId}>{stockInfo.fullName}</option>
+			);
+        
+        }
+        console.log(this.props.currentStockId);
```
Comment (ishanisri) [inline public/src/components/admin/Dividend.tsx]: Remove console.log

File: public/src/components/admin/Dividend.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        return (
+            <React.Fragment>
+            <div>
+             <table className="ui inverted table unstackable" style={{maxWidth: "600px", minWidth:"300px"}}>
+             <tbody>
+              <tr>
+                <td>
+                   <select className="stock d-inline ui fluid search selection dropdown" name="dividend-stock" value={this.props.currentStockId.toString()} onChange={this.handleStockIdChange.bind(this)}>
```
Comment (ishanisri) [inline public/src/components/admin/Dividend.tsx]: Check the styling in trading terminal for this selection box and use the same.

File: public/src/components/admin/Dividend.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+				    <i className="dropdown icon"></i>
+                      {options}
+                   </select>
+                </td>
+                <td>
+                  <input type="integer" className="market-input" id="dividend-amount" name="dividend-amount" onChange={this.handleDividendAmountChange.bind(this)} value={this.props.dividendAmount.toString()}/>
+                </td>
+                <td>
+                 <input type="button" className="ui inverted green button" onClick={this.props.applyDividendCallback.bind(this)} value="Divide Now"/>
```
Comment (ishanisri) [inline public/src/components/admin/Dividend.tsx]: Would be more appropriate to write `Send Dividends` as the text in the button.

**Pull request closed without merging**